docs/general/how-ra-works.md

```diff
@@ -0,0 +1,45 @@
+
+### How Is Achievement Code Processed?
+
+A game’s RAM is its most basic form.  RAM is simply the game's memory addresses and their respective values at any given time.  When processed by a console or emulator, the RAM can be used to recognize what is happening within the game.  Games are processed in a series of frames, generally 60, 50, 30 or 25 per second depending on console and format.  RA processes all unearned achievement logic in a set every frame.  Because of this, developers can create logic such that when all of its conditions are simultaneously true, identify something uniquely precise happening in a game and award an achievement, activate a leaderboard and provide accurate Rich Presence.  It is an achievement developer’s job to understand enough of a game's RAM to be able to construct this logic.
+
+Understanding how the RAM works that a developer intends to use in their logic is essential to creating a stable set that awards achievements, submits leaderboard scores and accurately displays Rich Presence only as intended.  RA developers use developer emulator tools to inspect a game's RAM and determine which addresses are responsible for things they intend to use for logic.  Developers have many ways to precisely define exactly which memory conditions must be true in order to construct logic, including retaining knowledge that something occurred previously in memory, but may no longer be true.
+
+### Achievement Logic and Processing Example
+
+As a simple example, to award an achievement for obtaining the hammer in Zelda II: The Adventure of Link, a developer would need to figure out which memory addresses within the game were associated with a few unique things about obtaining the hammer.  There are often many ways to do this.  One way a developer might approach this achievement is to find the addresses for the room ID to ensure the player is in the room with the hammer, also perhaps an area ID to ensure the player is in the right section of the game and lastly an address that indicates whether the player actually gets the hammer.  The reason a room and area ID are important is because when a player loads a save file that has collected the hammer, the memory would indicate that possessed of the hammer goes from untrue to true, but this isn't where the achievement should award.  The room and area ID conditions would ensure that a player was actually obtaining the hammer in game and not just loading a save file that already possessed it.  A solid achievement could be constructed to award when the following is true simultaneously:
```

Comment:
> There are often many ways to do this.

nit: While grammatically correct, this is stylistically redundant. Recommend axing the word "often".

Reply:
Concur and updated.